### ISSUE TITLE
AL09: False positive on T-SQL assignment alias syntax (Alias = Expression)

### ISSUE THREAD

**Issue by fank** [sqlfluff/sqlfluff#7544]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Rule AL09 (self-alias) flags T-SQL assignment aliases as "ambiguous self alias" when the alias name matches the source column name with a different casing. In T-SQL, `Alias = Table.column` is the idiomatic column aliasing syntax — it is **not** a self-alias.

For example, `ID = Orders.id` creates an alias `ID` for the expression `Orders.id`. This is semantically equivalent to `Orders.id AS ID` in standard SQL. AL09 incorrectly treats this as a self-alias because the alias text (`ID`) matches the column name (`id`) case-insensitively.

### Expected Behaviour

The following query should pass AL09 without violations:

```sql
SELECT
    ID = Orders.id
    ,CustomerName = Orders.customer_name
FROM
    dbo.Orders AS Orders
```

`ID = Orders.id` is a T-SQL assignment alias (alias on the left, expression on the right). It is not aliasing `id` as `id`.

### Observed Behaviour

```
L:   2 | P:   2 | AL09 | Ambiguous self alias. Either remove unnecessary alias,
                       | or quote alias/reference to make case change explicit.
                       | [aliasing.self_alias.column]
```

AL09 compares the alias name `ID` to the column name `id` case-insensitively and concludes it's a self-alias. But in T-SQL, `Alias = expression` is the standard way to alias columns — it is equivalent to `expression AS Alias`, not a redundant self-reference.

### How to reproduce

**test.sql:**
```sql
SELECT
    ID = Orders.id
    ,CustomerName = Orders.customer_name
FROM
    dbo.Orders AS Orders
```

```bash
sqlfluff lint --rules AL09 --dialect tsql test.sql
```

Output:
```
== [test.sql] FAIL
L:   2 | P:   2 | AL09 | Ambiguous self alias. Either remove unnecessary alias,
                       | or quote alias/reference to make case change explicit.
                       | [aliasing.self_alias.column]
```

### Dialect

tsql

### Version

sqlfluff 4.0.4, Python 3.10.12

### Configuration

```ini
[sqlfluff]
dialect = tsql
templater = raw
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
@peterbud @fank (Including @peterbud because he's put together a fix here).

I get that `ID = Orders.id` is idiomatic, the question is whether it's necessary in this context. My understanding is that unless configured otherwise, TSQL is not case sensitive, and so `ID = Orders.id` is unnecessary, you can just write `Orders.id` and have the same effect.

Correct me if I'm wrong though.

**Comment by fank** [comment]
I am using golang by calling T-SQL so even if T-SQL is case insensitive, the golang side would ran into issues because this is case sensitive.
This is why I would see this as false positive.

**Comment by peterbud** [comment]
> [@peterbud](https://github.com/peterbud) [@fank](https://github.com/fank) (Including [@peterbud](https://github.com/peterbud) because he's put together a fix here).
> 
> I get that `ID = Orders.id` is idiomatic, the question is whether it's necessary in this context. My understanding is that unless configured otherwise, TSQL is not case sensitive, and so `ID = Orders.id` is unnecessary, you can just write `Orders.id` and have the same effect.
> 
> Correct me if I'm wrong though.

Good observation, I agree. I'll close the PR.
@fank if that's an issue for you in your specific go environment, you can always disable a specific rule for a specific line/file, see it here: https://docs.sqlfluff.com/en/stable/configuration/ignoring_configuration.html
